pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2135,3 +2169,47 @@ func addAppInstanceOverlayRoute(
 			status.DisplayName)
 	}
 }
+
+// getSSHPasswords : returns decrypted SSH passwords
+func getSSHPasswords(ctx *zedrouterContext,
+	dc *types.AppNetworkConfig) ([]string, error) {
+	// TBD: add ssh keys into cypher block
+	return nil, nil
+}
+
+// getCloudInitUserData : returns decrypted cloud-init user data
+func getCloudInitUserData(ctx *zedrouterContext,
+	dc *types.AppNetworkConfig) (string, error) {
+	if dc.CipherBlockStatus.IsCipher {
+		status, decBlock, err := cipher.GetCipherCredentials(&ctx.decryptCipherContext,
+			agentName, dc.CipherBlockStatus)
+		ctx.pubCipherBlockStatus.Publish(status.Key(), status)
+		if err != nil {
+			log.Errorf("%s, appnetwork config cipherblock decryption unsuccessful, falling back to cleartext: %v",
+				dc.Key(), err)
+			decBlock.ProtectedUserData = *dc.CloudInitUserData
+			// We assume IsCipher is only set when there was some
+			// data. Hence this is a fallback if there is
+			// some cleartext.
+			if decBlock.ProtectedUserData != "" {
+				cipher.RecordFailure(agentName,
```

Comment:
I don't see publish of these counters, nor a subscribe of the in zedagent. I think that is how it is done for the other uses.

Reply:
Sorry, was not so familiar with this logic, updated.